src/batch-client/index.ts

```diff
@@ -595,7 +595,56 @@ export class ZimbraBatchClient {
 		return this.jsonRequest({
 			name: 'GetFolder',
 			body: denormalize(GetFolderRequestEntity)(options)
-		}).then(normalize(Folder));
+		}).then(res => {
+			const foldersResponse = normalize(Folder)(res);
+			const folders = get(foldersResponse, 'folders.0', {});
+
+			if (folders.linkedFolders) {
+				/**
+				 * This function is required because the API returns Subfolder data for shared folder
+				 * with Actual folder path (not mounted folder path). This could lead to 404 "NO SUCH FOLDER EXISTS ERROR".
+				 */
+				const updateAbsoluteFolderPath = (
+					originalName: any,
+					parentFolderAbsPath: string,
+					folders: any
+				) =>
+					folders.map((folder: any) => {
+						folder.absFolderPath = folder.absFolderPath.replace(
+							`/${originalName}`,
+							parentFolderAbsPath
+						);
+
+						if (folder.folders) {
+							folder.folders = updateAbsoluteFolderPath(
+								originalName,
+								parentFolderAbsPath,
+								folder.folders
+							);
+						}
+
+						return folder;
+					});
+
+				folders.linkedFolders = folders.linkedFolders.map((folder: any) => {
+					if (folder.view === FolderView.Message) {
```

Comment:
don't we need this for other message types?

Reply:
branch 3042 is created from 2589. This code is added in 2589.
This code is not added by me.

Reply:
@silentsakky Folders will have message/contact/appointment views. Are there any more ?

Reply:
below is the list of valid values from api docs
`|search folder|tag|conversation|message|contact|document|appointment|virtual conversation|remote folder|wiki|task|chat`